### PR TITLE
[lldb] Negate `is_signed` variable for argument `isUnsigned` in TypeSystemClang.cpp

### PR DESCRIPTION
**Pull request by kuilpd**

### CODE REVIEW

**Comment by kuilpd** [comment]
This change doesn't really affect anything in lldb yet, but will be used and tested afterwards in #115005

**Review by Michael137 — COMMENTED** [review]
LGTM

Can we add a test for this in `lldb/unittests/Symbol/TestTypeSystemClang.cpp`? We could call this API and then use `EnumConstantDecl::getInitVal` to retrieve the `APSInt` and make sure it's unsigned

File: lldb/unittests/Symbol/TestTypeSystemClang.cpp [hunk 1]
```diff
@@ -313,6 +313,16 @@ TEST_F(TestTypeSystemClang, TestGetEnumIntegerTypeBasicTypes) {
   }
 }
 
+TEST_F(TestTypeSystemClang, TestEnumerationValueSign) {
+  CompilerType enum_type = m_ast->CreateEnumerationType(
+      "my_enum_signed", m_ast->GetTranslationUnitDecl(),
+      OptionalClangModuleID(), Declaration(),
+      m_ast->GetBasicType(lldb::eBasicTypeSignedChar), false);
+  auto *enum_decl = m_ast->AddEnumerationValueToEnumerationType(
+      enum_type, nullptr, "minus_one", -1, 8);
```
Comment (Michael137) [inline lldb/unittests/Symbol/TestTypeSystemClang.cpp:322]: Nit:
```suggestion
  auto *enum_decl = m_ast->AddEnumerationValueToEnumerationType(
      enum_type, Declaration(), "minus_one", -1, 8);
```

**Review by Michael137 — APPROVED** [review]
LGTM, thanks! (one minor nit in the test)

**Pull request merged**